### ISSUE TITLE
Email functionality enhancement.

### ISSUE THREAD

**Issue by leftyb** [RestComm/RestComm-Core#831]
At the moment email functionality is very limited, meaning that it is not possible to actually use any kind of design pattern (e.g. html). We need to improve that, so user can actually structure the  email message as wanted (e.g. html, attachments, add photos, etc).

**Comment by tamcv** [comment]
Hi Jean,

As I understand, the mentioned email functionality is email verb in RVD. So far it just sends text message, so that we want it is able to send attachments also with this enhancement, right? 

Thanks,
Tam Chau

**Comment by deruelle** [comment]
That's correct. One addition though is that it's not limited to the RVD but also to the underlying Email API and potentially RCML, see http://docs.telestax.com/restcomm-pages/

**Comment by gvagenas** [comment]
Hi @tamcv did you had time to investigate the issue? Will you be able to take over this? Let me know if I can help.

George

**Comment by tamcv** [comment]
Thanks @gvagenas for your care. I have done some code changes for this issue, and now I get the problem that my "attach" field is not included in application xml file after saving. This is very first time I work on dynamic website, it's hard to understand RVD quickly as expected. :(
I have tried to push the changes via feature branch into RestComm repo  but it was failed due to permission denied. So I attach them here, could you please take a look on them and help me get out this situation?

Thanks so much.
Tam Chau

[831diff.zip](https://github.com/RestComm/RestComm-Core/files/132844/831diff.zip)

**Comment by tamcv** [comment]
Hi @gvagenas 

I have just pushed the changes for easily review. It cann't generate application xml file and I still keep on investigating further. Thanks

[https://github.com/RestComm/RestComm-Core/commit/44ade227bef0be97b4d5b026826b723a7f6ad58c](url)

**Comment by deruelle** [comment]
@tamcv can you read the Open Source Playbook Development Process Section again at https://docs.google.com/document/d/1RZz2nd2ivCK_rg1vKX9ansgNF6NpK_PZl81GxZ2MSnM/edit#heading=h.mcjitemt6ng1 and notably the parts on cloning the restcomm core repo and how to do pull requests

**Comment by tamcv** [comment]
Thanks @deruelle, actually I have not created "pull request" because the issue have not solved yet. I just pushed the change into my local branch to asked help about how to catch value of attribute in RVD verb (Email verb).

Thanks,
Tam Chau

**Comment by tamcv** [comment]
Hi all,

I have just implemented "sending files" feature successfully for Email API.
Perhaps I need more time for RCML and RVD. I will update status for this issue once finding any new.

**Comment by deruelle** [comment]
Thanks for updating us regularly on progress @tamcv

**Comment by gvagenas** [comment]
@tamcv you can also login to the Restcomm IRC - http://www.telestax.com/open-source/irc-channel/ - and ask your question or post your question to the public forum https://groups.google.com/forum/#!forum/restcomm
